UPGRADE_GUIDE.md

```diff
@@ -45,6 +47,23 @@ public ComponentType[] MinimumComponentTypes { get; } =
 
 > You will need to add `using Unity.Entities;` to the top of the file and reference it in the assembly that your custom GameObject creator is in.
 
+#### `void Register()` method
```

Comment:
It may be worth explaining the motivation for this given it seems a little odd otherwise